servers/src/main/java/tachyon/worker/block/evictor/LRFUEvictor.java

```diff
@@ -0,0 +1,345 @@
+    candidateDirView = dirCandidates.candidateDir();
+    List<Long> candidateBlocks = dirCandidates.candidateBlocks();
+    List<StorageTierView> tierViewsBelow =
+        mManagerView.getTierViewsBelow(candidateDirView.getParentTierView().getTierViewAlias());
+    // find a dir in below tiers to transfer blocks there, from top tier to bottom tier
+    StorageDirView candidateNextDir = null;
+    for (StorageTierView tierView : tierViewsBelow) {
+      candidateNextDir =
+          cascadingEvict(dirCandidates.candidateSize() - candidateDirView.getAvailableBytes(),
+              BlockStoreLocation.anyDirInTier(tierView.getTierViewAlias()), plan, sortedCRF);
+      if (candidateNextDir != null) {
+        break;
+      }
+    }
+    if (candidateNextDir == null) {
+      // nowhere to transfer blocks to, so evict them
+      plan.toEvict().addAll(candidateBlocks);
+    } else {
+      BlockStoreLocation dest = candidateNextDir.toBlockStoreLocation();
+      for (long block : candidateBlocks) {
+        plan.toMove().add(new Pair<Long, BlockStoreLocation>(block, dest));
+      }
+    }
+    return candidateDirView;
+  }
+
+  @Override
+  public EvictionPlan freeSpaceWithView(long bytesToBeAvailable, BlockStoreLocation location,
+      BlockMetadataManagerView view) {
+    synchronized (mBlockIdToLastAccessTime) {
+      updateCRFValue();
+      mManagerView = view;
+
+      List<Map.Entry<Long, Double>> sortedCRF = getSortedCRF();
+      List<Pair<Long, BlockStoreLocation>> toMove =
+          new ArrayList<Pair<Long, BlockStoreLocation>>();
+      List<Long> toEvict = new ArrayList<Long>();
+      EvictionPlan plan = new EvictionPlan(toMove, toEvict);
+      StorageDirView candidateDir = cascadingEvict(bytesToBeAvailable, location, plan, sortedCRF);
+
+      if (candidateDir == null) {
+        return null;
+      }
+
+      return plan;
+    }
+  }
+  
+  /**
+   * Sort all blocks in ascending order of CRF
+   * 
+   * @return the sorted CRF of all blocks
+   */
+  private List<Map.Entry<Long, Double>> getSortedCRF() {
+    List<Map.Entry<Long, Double>> sortedCRF = 
+        new ArrayList<Map.Entry<Long, Double>>(mBlockIdToCRFValue.entrySet());
+    Collections.sort(sortedCRF, new Comparator<Map.Entry<Long, Double>>() {
+      public int compare(Entry<Long, Double> o1, Entry<Long, Double> o2) {
+        double res = o1.getValue() - o2.getValue();
+        if (res < 0) {
+          return -1;
+        } else if (res > 0) {
+          return 1;
+        } else {
+          return 0;
+        }
+      }
+    });
+    return sortedCRF;
+  }
+
+  /**
+   * Calculate function F(t) = pow (1.0 / {@link #mAttenuationFactor}, t * {@link #mStepFactor})
+   * 
+   * @param logicTimeInterval time interval since that access to current
+   * @return Function value of F(t)
+   */
+  private double calculateFunction(long logicTimeInterval) {
+    return 1.0 * Math.pow(1.0 / mAttenuationFactor, 1.0 * logicTimeInterval * mStepFactor);
```

Comment:
Same, no need for `1.0 * logicTimeInterval * mStepFactor` since `mStepFactor` is `double`.